api_server.py

```
@@ -150,19 +165,25 @@ def total_hrs(p):
             "ba_non_ttr": 0,
         }
-    with get_db() as conn:
-        placeholders = ",".join("?" * len(available_ids))
+
+    # Support passing IDs (backward compatibility) or pre-fetched rows
+    if available_crew and isinstance(available_crew[0], int):
+        conn = get_db()
+        placeholders = ",".join("?" * len(available_crew))
         rows = conn.execute(
-            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})", available_ids
+            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})",
```

Ruff (0.15.7)

[error] Possible SQL injection vector through string-based query construction

(S608)